thresher/models.py

```diff
@@ -88,26 +88,47 @@ class Topic(models.Model):
     # an id within the given Analysis Type
     topic_id = models.IntegerField() 
 
-    # The analysis type to which this topic belongs
-    analysis_type = models.ForeignKey(AnalysisType, related_name='topics')
+    # an id of its parent topic
+    parent = models.ForeignKey("self", related_name="subtopics", on_delete=models.CASCADE)
 
     # The name of the topic
     name = models.TextField()
+
+    # Glossary related to the topic under analysis
+    glossary = models.TextField()                 # as a JSON map
+
+    instructions = models.TextField()
+
     
     class Meta:
-        unique_together = ("topic_id", "analysis_type")
+        unique_together = ("topic_id", "name") # not sure if this is correct.
```

Comment:
Actually, I don't even think topic_id makes sense anymore, unless we are interested in the "order" of topics in a hierarchy. @nickbadams is the order of subtopics of a given topic important? If yes, then I would say we need a unique_together = ("topic_id", "parent"), that would be equivalent to the old case.
As for the name, Nick could maybe give us an idea of what name uniqueness constraints, if any, we should impose.